### PR TITLE
Tag existing events with experimentation info in typescript-language-features

### PR DESCRIPTION
**Pull request by DanielRosenwasser**

### CODE REVIEW

File: extensions/typescript-language-features/src/utils/packageInfo.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import * as vscode from 'vscode';
+
+export interface PackageInfo {
+	name: string;
+	version: string;
+	aiKey: string;
+}
+
+export function getPackageInfo(context: vscode.ExtensionContext) {
```
Comment (DanielRosenwasser) [inline extensions/typescript-language-features/src/utils/packageInfo.ts:14]: Note this is slightly different from the old `packageInfo` getter.

File: extensions/typescript-language-features/src/extension.ts [hunk 2]
```diff
@@ -42,6 +45,19 @@ export function activate(
...
 
+	let experimentTelemetryReporter: IExperimentationTelemetryReporter | null = null;
+	const packageInfo = getPackageInfo(context);
+	if (packageInfo) {
+		const { name: id, version, aiKey } = packageInfo;
+		const vscTelemetryReporter = new VsCodeTelemetryReporter(id, version, aiKey);
+		experimentTelemetryReporter = new ExperimentationTelemetryReporter(vscTelemetryReporter);
+		context.subscriptions.push(experimentTelemetryReporter);
+
+		// Don't bother creating an experimentation service if we don't have a valid reporter.
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/extension.ts:56]: Not sure what this comment means. Isn't the `ExperimentationService` always created?

File: extensions/typescript-language-features/src/typeScriptServiceClientHost.ts [hunk 3]
```diff
@@ -72,6 +73,7 @@ export default class TypeScriptServiceClientHost extends Disposable {
 			processFactory: TsServerProcessFactory;
 			activeJsTsEditorTracker: ActiveJsTsEditorTracker;
 			serviceConfigurationProvider: ServiceConfigurationProvider;
+			experimentTelemetryReporter: IExperimentationTelemetryReporter | null;
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/typeScriptServiceClientHost.ts:76]: [nit] Please switch this use `undefined` instead

**Review by mjbvz — CHANGES_REQUESTED** [review]

File: extensions/typescript-language-features/src/extension.ts (hunk 2 above)
Comment (DanielRosenwasser) [reply extensions/typescript-language-features/src/extension.ts]: I have also made removed disposal from the `TelemetryReporter` in `extensions/typescript-language-features/src/utils/telemetry.ts` as well.

**Review by mjbvz — APPROVED** [review]
Thanks!

**Pull request merged**